### PR TITLE
[15.0][IMP] stock_picking_product_assortment: use base_view_inheritance_extension module to extend product domain. Unnecessary code removed

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT44966

depends on:
- [x] https://github.com/OCA/server-tools/pull/2706

ping @chienandalu @pilarvargas-tecnativa

### CODE REVIEW

**Comment by pilarvargas-tecnativa** [comment]
@sergio-teruel rebase please

**Comment by sergio-teruel** [comment]
Rebase done!

**Review by pilarvargas-tecnativa — APPROVED** [review]
Tested locally, thanks!

Please remove the dependency commit

**Review by chienandalu — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@sergio-teruel can you check the CI error?

File: stock_picking_product_assortment/__manifest__.py [hunk 1]
```diff
@@ -10,6 +10,6 @@
     "maintainers": ["CarlosRoca13"],
     "license": "AGPL-3",
     "installable": True,
-    "depends": ["stock", "product_assortment"],
+    "depends": ["stock", "product_assortment", "base_view_inheritance_extension"],
```
Comment (rousseldenis) [inline stock_picking_product_assortment/__manifest__.py]: Oooh, you add a new dependency, great ! :wink:

Comment (sergio-teruel) [reply stock_picking_product_assortment/__manifest__.py]: Yeah!! I know that is not a good idea but this way I remove a module to be maintained

Comment (pedrobaeza) [reply stock_picking_product_assortment/__manifest__.py]: @sergio-teruel please sort them alphabetically.

@rousseldenis is it OK for you then?

Comment (rousseldenis) [reply stock_picking_product_assortment/__manifest__.py]: For me, improvements even with several added modules in dependencies is always good :stuck_out_tongue_winking_eye:

**Comment by sergio-teruel** [comment]
@pedrobaeza Changes done!

**Pull request merged**